code/modules/hydroponics/seed_extractor.dm

```diff
@@ -97,7 +117,20 @@
 	if(!isliving(user) || user.combat_mode)
 		return ..()
 
-	if(default_deconstruction_screwdriver(user, "sextractor_open", "sextractor", attacking_item))
+	if( \
+		default_deconstruction_screwdriver( \
+			user, \
+			icon_state_open = \
+				powered() ? \
+					(machine_stat & BROKEN ? "sextractor-broken-open" : "sextractor_open") : \
+					(machine_stat & BROKEN ? "sextractor-broken-off-open" : "sextractor-off-open"), \
+			icon_state_closed = \
+				powered() ? \
+					(machine_stat & BROKEN ? "sextractor-broken" : "sextractor") : \
+					(machine_stat & BROKEN ? "sextractor-broken-off" : "sextractor-off"), \
+			screwdriver = attacking_item \
+		) \
+	)
```

Comment:
You don't *need* to have `default_deconstruction_screwdriver` handle the icon state updates.
You could just run `default_deconstruction_screwdriver(user = user, screwdriver = attacking_item)` and then call `update_appearance(UPDATE_ICON_STATE)`, and it'll hand everything for you.